sklearn/decomposition/tests/test_pca.py

```diff
@@ -568,3 +568,62 @@ def test_pca_n_components_mostly_explained_variance_ratio():
     n_components = pca1.explained_variance_ratio_.cumsum()[-2]
     pca2 = PCA(n_components=n_components).fit(X, y)
     assert pca2.n_components_ == X.shape[1]
+
+
+def test_infer_dim_bad_spec():
+    # Test a spectrum that drops to near zero
+    spectrum = np.array([1, 1e-30, 1e-30, 1e-30])
+    n_samples = 10
+    n_features = 5
+    ret = _infer_dimension_(spectrum, n_samples, n_features)
+    assert ret == 0
+
+
+def test_assess_dimension_error_rank_greater_than_features():
+    # Test error when tested rank is greater than the number of features
+    spectrum = np.array([1, 1e-30, 1e-30, 1e-30])
+    n_samples = 10
+    n_features = 4
+    rank = 5
+    with pytest.raises(ValueError):
+        ret = _assess_dimension_(spectrum, rank, n_samples, n_features)
+
+
+def test_assess_dimension_same_n_rank_and_features():
+    # Test that
+    spectrum = np.array([1, 1e-30, 1e-30, 1e-30])
+    n_samples = 10
+    n_features = 4
+    rank = 4
+    ret = _assess_dimension_(spectrum, rank, n_samples, n_features)
+    assert ret is not None
+
+
+def test_assess_dimension_small_eigenvalues():
+    # Test tiny eigenvalues appropriately when 'mle'
+    spectrum = np.array([1, 1e-30, 1e-30, 1e-30])
+    n_samples = 10
+    n_features = 5
+    rank = 4
+    ret = _assess_dimension_(spectrum, rank, n_samples, n_features)
+    assert ret == -np.inf
+
+
+def test_infer_dim_mle():
+    # Test small eigenvalues when 'mle' with pathelogical 'X' dataset
+    X, _ = datasets.make_classification(n_informative=1, n_repeated=18,
+                                        n_redundant=1, n_clusters_per_class=1,
+                                        random_state=42)
+    pca = PCA(n_components='mle').fit(X)
+    assert pca.n_components_ == 0
+
+
+def test_fit_mle_too_few_samples():
+    # Tests that an error is raised when the number of samples is smaller
+    # than the number of features during an mle fit
+    X, _ = datasets.make_classification(n_samples=20, n_features=21,
+                                        random_state=42)
+
+    pca = PCA(n_components='mle', svd_solver='full')
+    with pytest.raises(ValueError):
```

Comment:
you should check (even partially) the error message raised. You can use `pytest.raises(ValueError, match=err_msg)` where `err_msg` can be a (meaningful) part of the error message.